### PR TITLE
add hasher pool

### PR DESCRIPTION
**Pull request by acud**

### CODE REVIEW

File: pool/pool.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"golang.org/x/crypto/sha3"
+)
+
+// HasherPool pools bmt Hashers.
+// It provides the ability for the number of hashers to grow
+// according to demand, but will shrink once the minimum defined
+// hashers are put back into the pool.
+type HasherPool interface {
```
Comment (janos) [inline pool/pool.go]: I would maybe call this just `Interface`.

File: pool/pool.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	// Get a bmt Hasher instance.
+	// Instances are reset before being returned to the caller.
+	Get() *bmtlegacy.Hasher
+	// Put a bmt Hasher back into the pool
+	Put(*bmtlegacy.Hasher)
+	// Size of the pool.
+	Size() int
+}
+
+type hasherPool struct {
```
Comment (janos) [inline pool/pool.go]: Maybe to call this just `pool`?

File: pool/pool.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
+package pool
```
Comment (janos) [inline pool/pool.go:5]: Add copy.

File: pool/pool_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package pool_test
```
Comment (janos) [inline pool/pool_test.go:5]: Add copy.

**Review by janos — APPROVED** [review]
Just a few suggestions.

**Comment by acud** [comment]
cool. any idea why the ci is now failing again?

File: pool/pool.go (hunk 1 above)
Comment (acud) [reply pool/pool.go]: i'll change to `Pooler`. Not really a huge fan of using the `<package>.Interface` convention tbh

**Comment by janos** [comment]
> cool. any idea why the ci is now failing again?

gitlabci-lint is failing to build. It is better to install it with this command https://github.com/ethersphere/bee/blob/master/Makefile#L34.

**Comment by acud** [comment]
@janos it works on the ubuntu and macos build but fails on the beloved windows: https://github.com/ethersphere/bmt/pull/11/checks?check_run_id=1218969427

**Comment by acud** [comment]
switched to the official golangci github action and it works

**Pull request merged**